src/main/java/graphql/schema/idl/SchemaDirectiveWiringSchemaGeneratorPostProcessing.java

```diff
@@ -0,0 +1,133 @@
+import graphql.schema.GraphQLInterfaceType;
+import graphql.schema.GraphQLNamedType;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLScalarType;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.GraphQLSchemaElement;
+import graphql.schema.GraphQLTypeVisitorStub;
+import graphql.schema.GraphQLUnionType;
+import graphql.schema.SchemaTransformer;
+import graphql.util.TraversalControl;
+import graphql.util.TraverserContext;
+import graphql.util.TreeTransformerUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import static graphql.util.TraversalControl.CONTINUE;
+
+@Internal
+class SchemaDirectiveWiringSchemaGeneratorPostProcessing implements SchemaGeneratorPostProcessing {
+
+    private final SchemaGeneratorDirectiveHelper generatorDirectiveHelper = new SchemaGeneratorDirectiveHelper();
+    private final TypeDefinitionRegistry typeRegistry;
+    private final RuntimeWiring runtimeWiring;
+    private final GraphQLCodeRegistry.Builder codeRegistryBuilder;
+    private final Map<String, Object> directiveBehaviourContext = new HashMap<>();
+
+
+    public SchemaDirectiveWiringSchemaGeneratorPostProcessing(TypeDefinitionRegistry typeRegistry, RuntimeWiring runtimeWiring, GraphQLCodeRegistry.Builder codeRegistryBuilder) {
+        this.typeRegistry = typeRegistry;
+        this.runtimeWiring = runtimeWiring;
+        this.codeRegistryBuilder = codeRegistryBuilder;
+    }
+
+
+    @Override
+    public GraphQLSchema process(GraphQLSchema originalSchema) {
+        GraphQLSchema newSchema = SchemaTransformer.transformSchema(originalSchema, new Visitor());
+        return newSchema.transform(builder -> {
+            // they could have changed the code registry so rebuild it
+            GraphQLCodeRegistry codeRegistry = this.codeRegistryBuilder.build();
+            builder.codeRegistry(codeRegistry);
+        });
+    }
+
+    public class Visitor extends GraphQLTypeVisitorStub {
+
+        private SchemaGeneratorDirectiveHelper.Parameters mkBehaviourParams() {
+            return new SchemaGeneratorDirectiveHelper.Parameters(typeRegistry, runtimeWiring, directiveBehaviourContext, codeRegistryBuilder);
+        }
+
+        private TraversalControl changOrContinue(GraphQLSchemaElement node, GraphQLSchemaElement newNode, TraverserContext<GraphQLSchemaElement> context) {
+            if (node != newNode) {
+                TreeTransformerUtil.changeNode(context, newNode);
+            }
+            return CONTINUE;
+        }
+
+        private boolean isIntrospectionType(GraphQLNamedType type) {
+            return type.getName().startsWith("__");
+        }
+
+        private <T extends GraphQLNamedType> boolean notSuitable(T node, Function<T, NamedNode<?>> suitableFunc) {
+            if (isIntrospectionType(node)) {
+                return true;
+            }
+            NamedNode<?> definition = suitableFunc.apply(node);
```

Comment:
the schema directive wiring code makes an assumption we have a SDL AST definition behind it - so it's SDL only

Reply:
do we need to restrict it? Is it bad that we restrict it?

Reply:
We don't want any one tweaking introspection types I think